scalr/data_source_environment.go

```diff
@@ -85,6 +111,11 @@ func dataSourceEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("account_id", environment.Account.ID)
 	d.Set("cost_estimation_enabled", environment.CostEstimationEnabled)
 	d.Set("status", environment.Status)
+	if envID != "" {
+		d.SetId(envID)
+	} else {
+		d.SetId(environment.ID)
+	}
```

Comment:
I think, this `if` is not necessary here, in both cases you still could use `d.SetId(environment.ID)`

Reply:
And move `SetID` to the end of the function, for code consistency